### ISSUE TITLE
Query on Using Validation Set as Test Set in nnUNet Framework

### ISSUE THREAD

**Issue by siny1998** [MIC-DKFZ/nnUNet#2394]
Hi,

Great work on the project! I have a question concerning nnUNet's methodology. Since nnUNet consistently uses the checkpoint from the last iteration as the final model weights, would it be feasible to adjust the data split such that the test data acts as the validation set? As we are aware, a validation process is conducted after the training process. Can we directly use the outcomes of this validation as the test data results?

Best regards,
Siny

**Comment by constantinulrich** [comment]
Hi, you can manually change the dataset_split.json however you want. 
But in most cases you don't want to evaluate on your testset because of good scientific practices... 
I would recommend to perform the 5fold cross-validation to achieve stable results on small datasets, and if possible (if the dataset is large enough) have an untouched test set. 
Best Constantin

**Comment by constantinulrich** [comment]
Hi, 
technically it would work and if you do as you described and train only once it would not break anything. 
However, during training the val and pseudo dice curves are done on the test set and the "best_checkpoint"  (which is not used per default) is as well determined on the test set then. So be careful. 

-->  Why not perform a five-fold cross validation on the train set, and predict the assemble on the test set (gives your better results anyway...)

**Comment by siny1998** [comment]
Hi Constantin,

Thank you very much for your guidance. I plan to use the "final checkpoint" not "best checkpoint" for evaluation, thereby avoiding any exposure to the test data during training.

I will certainly take your suggestion into consideration. Thank you once again for your invaluable advice.

Best regards,
Siny
